File: Desenvolvimento/client/src/@fuse/types/models/setor-endereco.ts
import { Endereco } from './endereco';
import { Setor } from './setor';
import { BaseModel } from '../base-model';

export class SetorEndereco extends BaseModel {
    constructor(
        public id: number,
        public setorId?: number,
        public setor?: Setor,
        public enderecoId?: number,
        public endereco?: Endereco
        ) { super(); }
}

File: Desenvolvimento/server/src/ApplicationCore/Entities/SetorEndereco.cs
using ApplicationCore.Entities.Identity;
using ApplicationCore.Interfaces.Base;
using System;
using System.Collections.Generic;
using System.Text;

namespace ApplicationCore.Entities
{
    public class SetorEndereco : IBaseEntity<long>
    {
        public long Id { get; set; }

        public long? SetorId { get; set; }
        public virtual Setor Setor { get; set; }

        public long? EnderecoId { get; set; }
        public virtual Endereco Endereco { get; set; }
    }
}
